src/Func.h

```diff
@@ -66,6 +89,63 @@ class Stage {
     /** Return the name of this stage, e.g. "f.update(2)" */
     EXPORT const std::string &name() const;
 
+    /** Calling rfactor() on an associative update definition a Func will split
+     * the update into an intermediate which computes the partial results and
+     * replace the current update definition with a new definition which merges
+     * the partial results. If called on a init/pure definition, this will
+     * throw an error. rfactor() will automatically infer the associative reduction
+     * operator and identity of the operator. If it can't prove the operation
+     * is associative or if it cannot find an identity for that operator, this
+     * will throw an error.
+     *
+     * rfactor() takes as input 'preserved', which is a list of <RVar, Var> pairs.
+     * The rvars not listed in 'preserved' are removed from the original Func and
+     * are lifted to the intermediate Func. The remaining rvars (the ones in
+     * 'preserved') are made pure in the intermediate Func. The intermediate Func's
+     * update definition will inherit all scheduling directives (e.g. split, etc.)
+     * applied to the original Func's update definition.
+     *
+     * For example, f.update(0).rfactor({{r.y, u}}) would rewrite a pipeline like this:
+     \code
+     f(x) = 0;
+     f(x) += g(r.x, r.y);
+     \endcode
+     * into a pipeline like this:
+     \code
+     f_intm(u, x) = 0;
+     f_intm(u, x) += g(r.x, u);
+
+     f(x) = 0;
+     f(x) = f_intm(r.y, x);
+     \endcode
+     *
+     * This has a variety of uses. You can use it to split computation of an associative reduction:
+     \code
+     f() = -inf;
+     RDom r(0, 96);
+     f() = max(f(), g(r.x));
+     f.update(0).split(r.x, rxo, rxi, 8);
+     f.update(0).rfactor({{rxo, u}}).compute_root().parallel(u);
+     \endcode
+     *
+     *, which is equivalent to:
+     \code
+     for u = 0 to 11:
```

Comment:
In the schedule it looks like only the pure definition of the intermediate is parallelized, but in the pseudocode only the update definition is parallelized. Maybe change the schedule to compute_root().parallel(u).update().parallel(u) and make both parallel?